packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -1603,6 +1605,36 @@ function updateHostResource(current, workInProgress, renderLanes) {
   return workInProgress.child;
 }
 
+function updateHostSingleton(
+  current: Fiber | null,
+  workInProgress: Fiber,
+  renderLanes: Lanes,
+) {
+  pushHostContext(workInProgress);
+
```

Comment:
In the complete work phase HostSingletons hit the HostComponent path on update renders. I was going to try to do something similar here but the interaction between single text children and the placement of child fibers beneath is wonky. For now I'll leave it like this which may actually be desirable